internal/venice-common/src/main/java/com/linkedin/venice/throttle/VeniceRateLimiter.java

```diff
@@ -0,0 +1,26 @@
+package com.linkedin.venice.throttle;
+
+public interface VeniceRateLimiter {
+  enum RateLimiterType {
+    EVENT_THROTTLER_WITH_SILENT_REJECTION, GUAVA_RATE_LIMITER, TOKEN_BUCKET_INCREMENTAL_REFILL,
+    TOKEN_BUCKET_GREEDY_REFILL,
+  }
+
+  /**
+   * Try to acquire permit for the given rcu. Will not block if permit is not available.
+   * @param units Number of units to acquire.
+   * @return true if permit is acquired, false otherwise.
+   */
+  boolean tryAcquirePermit(int units);
+
+  /**
+   * Helper methods used
+   */
+  void setGlobalPermitsCount(long totalPermits);
+
+  void setThisNodeResponsibility(double thisNodeResponsibility);
+
+  long getGlobalPermitsCount();
+
+  double getThisNodeResponsibility();
```

Comment:
This feels a bit odd, why do the underlying rate limiter need to be aware of the node responsibility? If it's only used to determine if two rate limiter is the same (for deciding if we need to create a new rate limiter) then wouldn't be better to just have setter and getter on the post node responsibility permits count and use that for comparison instead of the global permits count + node responsibility?

Reply:
Originally added this to help aide debugging but wha you're saying is good enough too